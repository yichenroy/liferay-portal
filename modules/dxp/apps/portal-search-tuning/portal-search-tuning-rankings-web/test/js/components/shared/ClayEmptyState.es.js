/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import ClayEmptyState from '../../../../src/main/resources/META-INF/resources/js/components/shared/ClayEmptyState.es';
import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe('ClayEmptyState', () => {
	it('renders', () => {
		const {asFragment} = render(<ClayEmptyState />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('displays a custom title', () => {
		const {getByText} = render(<ClayEmptyState title="Test Title" />);

		expect(getByText('Test Title')).toBeInTheDocument();
	});

	it('displays a custom description', () => {
		const {getByText} = render(
			<ClayEmptyState description="Test Description" />
		);

		expect(getByText('Test Description')).toBeInTheDocument();
	});
});
