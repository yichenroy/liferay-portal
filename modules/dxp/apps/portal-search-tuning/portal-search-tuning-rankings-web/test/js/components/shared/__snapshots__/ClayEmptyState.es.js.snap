// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayEmptyState renders 1`] = `
<DocumentFragment>
  <div
    class="empty-state-root"
  >
    <img
      alt="Empty State Image"
      class="empty-state-image"
      src="/o/admin-theme/images/states/search_state.gif"
    />
    <div
      class="empty-state-title"
    >
      No Results Found
    </div>
    <div
      class="empty-state-description"
    >
      Sorry, no results were found.
    </div>
  </div>
</DocumentFragment>
`;
